templates/agent/client/components/chat-history/AgentChangeHistoryItems.tsx

```diff
@@ -0,0 +1,179 @@
+import { forwardRef, useCallback, useMemo } from 'react'
+import {
+	DefaultShapeWrapper,
+	Editor,
+	RecordsDiff,
+	reverseRecordsDiff,
+	TLRecord,
+	TLShape,
+	TLShapeId,
+	TLShapeWrapperProps,
+	useValue,
+} from 'tldraw'
+import { $chatHistoryItems } from '../../atoms/chatHistoryItems'
+import { AgentChangeHistoryItem } from '../../types/ChatHistoryItem'
+import TldrawViewer from './TldrawViewer'
+
+export function AgentChangeHistoryItems({
+	items,
+	editor,
+}: {
+	items: AgentChangeHistoryItem[]
+	editor: Editor
+}) {
+	const diffShapes = items.map((item) => getDiffShapesFromDiff({ diff: item.diff })).flat()
+
+	const handleAccept = useCallback(() => {
+		$chatHistoryItems.update((oldItems) => {
+			const newItems = [...oldItems]
+			for (const item of items) {
+				const index = newItems.findIndex((v) => v === item)
```

Comment:
i got pretty confused here because there are a lot of things called items and i didn't realise at first how different `items`/`oldItems` were from one and other. I'd be tempted to add some comments and maybe rename `items` to `itemsToDisplay` or something